hedera-cryptography/hedera-cryptography-tss/src/main/java/com/hedera/cryptography/tss/api/TssService.java

```diff
@@ -0,0 +1,167 @@
+import com.hedera.cryptography.tss.impl.TssServiceImpl;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.List;
+import java.util.Random;
+
+/**
+ * A Threshold Signature Scheme Service.
+ * Contract of TSS:
+ * <ul>
+ *     <li>Generate TssMessages out of PrivateShares</li>
+ *     <li>Verify TssMessages out of a ParticipantDirectory</li>
+ *     <li>Obtain PrivateShares out of TssMessages for each owned share</li>
+ *     <li>Aggregate PrivateShares</li>
+ *     <li>Obtain PublicShares out of TssMessages for each share</li>
+ *     <li>Aggregate PublicShares</li>
+ *     <li>Sign Messages</li>
+ *     <li>Verify Signatures</li>
+ *     <li>Aggregate Signatures</li>
+ * </ul>
+ * @implNote an instance of the service would require a source of randomness {@link java.util.Random}, and a{@link SignatureSchema}
+ */
+public interface TssService {
+
+    /**
+     * Generate a TSS message for a pendingParticipantDirectory, from a random private share.
+     *
+     * @param pendingParticipantDirectory the pending participant directory that we should generate the message for
+     * @return a TSSMessage produced out of a random share.
+     */
+    @NonNull
+    TssMessage generateTssMessage(@NonNull TssParticipantDirectory pendingParticipantDirectory);
+
+    /**
+     * Generate a TSS message for a pendingParticipantDirectory, for the specified {@link TssPrivateShare}.
+     *
+     * @param pendingParticipantDirectory the pending participant directory that we should generate the message for
+     * @param privateShare       the secret to use for generating new keys
+     * @return a TSSMessage for each owned share.
+     */
+    @NonNull
+    TssMessage generateTssMessages(
+            @NonNull TssParticipantDirectory pendingParticipantDirectory, @NonNull TssPrivateShare privateShare);
+
+    /**
+     * Verify that a {@link TssMessage} is valid.
+     *
+     * @param participantDirectory the participant directory used to generate the message
+     * @param tssMessages the TSS messages to validate
+     * @return true if the message is valid, false otherwise
+     */
+    boolean verifyTssMessage(@NonNull TssParticipantDirectory participantDirectory, @NonNull TssMessage tssMessages);
+
+    /**
+     * Compute all private shares that belongs to this participant from a threshold minimum number of {@link TssMessage}s.
+     * It is the responsibility of the caller to ensure that the list of validTssMessages meets the required threshold.
+     *
+     * @param participantDirectory the pending participant directory that we should generate the private share for
+     * @param validTssMessages the TSS messages to extract the private shares from. They must be previously validated.
+     * @return a sorted by sharedId list of private shares the current participant owns, or null if there aren't enough shares to meet the threshold.
```

Comment:
the rest of the doc says it's the responsibility of the caller to make sure it meets the threshold, yet this return doc claims it will return null if the threshold isn't met.

Reply:
I respected this caveat from the original draft. It is a valid warning as if saying if you don't want to get null, count the messages first. 
But let's discuss this as I agree with a comment you made later suggesting that we should throw an exception, with the note that we should be consistent.